### ISSUE TITLE
Symbiosis

### ISSUE THREAD

**Issue by JeffreyStrahm** [siggame/MegaMinerAI-Game-Ideas#61]
# **The Game**
## **TLDR**
- 160 by 85
- randomly growing vegetation
- some underwater
- all creatures die of old age.
- creatures drop meat when they die
- two teams of creatures

## **Story**
Two teams of scientists who have mastered evolution are having a competition to see who can make the best eco-system. Prove your team are better than the other and get permanent bragging rights over your co-workers!

## **Victory**
Simply put, be more successful than your opponent! Wither it be just having a larger, denser, ecosystem, or killing your opponent.
1. If your opponent died, you win
2. If you have more biomass, you win. (aka pop count + food in bellies)
3. If you have the healthiest population (most HP, age not counted)
4. otherwise coin flip.

## **Map**
The map is 160 by 85. Around the island, there is water with a few sparse lakes.
every tile has a vegetation value. These values grow by a set growth rate every turn (I would stay away from vegetation spread as it would be too complex for the player).

## **Evolution**
This is how a species changes from it's starting basic biomass to a complex organism with its own place in the ecosystem! Each creature has the following traits.

| Trait | Start | Cap | Min | cost (EP) | description |
|----|----|----|----|----|----|
|**Land Speed**|1|5|0|1|"How many tiles per turn your creature can move on land"|
|**Aquatic Speed**|1|5|0|1|"How many tiles per turn your creature can move in the water"|
|**Fat**|1|10|0|1|"The amount of food a creature can store in its belly"|
|**Claws**|0|4|0|2|"The amount of damage the creatures does when attacking"|
|**Reach**|0|3|1|4|"The distance at which this creature can interact"|
|**Vegetarian**|1|5|0|1|"The number of vegetation this creature can eat per turn."|
|**Carnivore**|0|5|0|1|"The number of meat this creature can eat per turn"|
|**Land Regen**|1|10|-5|4|"The amount of energy this creaturs regens when on land"|
|**Aquatic Regen**|-5|10|-5|4|"The amount of energy this creature regens when in the sea"|

Creatures gain 1 evolution points every x turns and they also gain the ability to refund one trait, lowering it a level and gaining its cost in EP. All creatures of a player evolve in the same way.

## **Creatures**
Every creature has a few properties:

Energy - this is used whenever a creature performs an action like moving and attacking. Regens based on the designated regen rate. If it drops below 0, the creatures die due to exhaustion/drowning/starvation. They do not regen if they didn't eat and had no food stored to compensate.
Fat - The amount of food the creature has in reserve.
Age - The current age of this creature. Once x turns has passed and it reaches the age limit, the creature dies and drops less meat than if it was killed.
Health - The amount of HP these creatures has. Regens every turn, bonus if eating. If it hits 0 or below, the creature dies. Drops more food if it has stored food in fat.
EP - the number of unspent evolution points.

**Actions:**
These are the actions creatures can take

Attack - does damage to the target according to the claw damage. Reach controls how far away you can target. Costs energy
Move - Moves the creature the number of tiles they can. Costs energy to move at all.
Breed - Create another of your species! (can only do this with another of your species, not even your teammates)
Eat - eats food within reach distance. The amount you eat is determined by your **Carnivore** and **Vegetarian** traits
